watt.py

```diff
@@ -172,6 +173,27 @@ def download_gcs_blob_to_tmp_file(self, blob_path: str, tmp_file_name: str) -> N
         blob = bucket.blob(blob_str)
         blob.download_to_filename(tmp_file_name)
 
+    @staticmethod
+    def read_in_chunks(file_object, chunk_size=1024*1024):
+        while True:
+            data = file_object.readlines(chunk_size)
+            if not data:
+                break
+            yield data
+
+    @staticmethod
+    def compare_files(file1: io.TextIOWrapper, file2: io.TextIOWrapper, line_skip_regex_str: str = None) -> int:
+        line_skip_regex = None
+        if line_skip_regex_str:
+            line_skip_regex = re.compile(line_skip_regex_str)
+        for chunk1, chunk2 in zip(CompareOutputs.read_in_chunks(file1), CompareOutputs.read_in_chunks(file2)):
```

Comment:
Are you sure this works in all edge cases? For example, suppose you want to strip a header-like line in one file like `#<current_date>` and the other doesn't have it. Because you're reading the files in fixed chunk sizes based on the byte-size, wouldn't the two get offset so that the `chunk1` and `chunk2` below might have different lengths? Maybe the regex filtering should happen in the generator created via `read_in_chunks` to avoid this?

Reply:
I also think if one file has a longer comment line than the other, you could read fewer lines from one than the other. So you'll likely need to filter, then compare each entry from the chunk list, or some similar idea to reimplement this.